### PR TITLE
Upgraded version of knife-azure azure-sdk libraries. 

### LINKED ISSUE #490: azure_mgmt_resources and  azure_mgmt_network conflict errors
# Version:

1.8.0

# Environment: [Details about the environment such as the Operating System, cookbook details, etc...]
Windows 10.
ChefDK: 3.3.23

# Steps to Reproduce:
knife azure image list

# Actual Result:

Traceback (most recent call last):
        18: from C:/opscode/chefdk/bin/knife:307:in `<main>'
        17: from C:/opscode/chefdk/bin/knife:307:in `load'
        16: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/bin/knife:24:in `<top (required)>'
        15: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/application/knife.rb:161:in `run'
        14: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife.rb:215:in `run'
        13: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife.rb:152:in `subcommand_class_from'
        12: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife/core/subcommand_loader.rb:109:in `command_class_from'
        11: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife/core/subcommand_loader.rb:95:in `load_command'
        10: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife/core/subcommand_loader.rb:85:in `load_commands'
         9: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife/core/subcommand_loader.rb:85:in `each'
         8: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife/core/subcommand_loader.rb:85:in `block in load_commands'
         7: from C:/opscode/chefdk/embedded/lib/ruby/gems/2.5.0/gems/chef-14.5.33-universal-mingw32/lib/chef/knife/core/subcommand_loader.rb:85:in `load'
         6: from C:/Users/fferre/AppData/Local/chefdk/gem/ruby/2.5.0/gems/knife-azure-1.8.0/lib/chef/knife/azurerm_base.rb:21:in `<top (required)>'
         5: from C:/opscode/chefdk/embedded/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
         4: from C:/opscode/chefdk/embedded/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:128:in `rescue in require'
         3: from C:/opscode/chefdk/embedded/lib/ruby/2.5.0/rubygems.rb:217:in `try_activate'
         2: from C:/opscode/chefdk/embedded/lib/ruby/2.5.0/rubygems.rb:224:in `rescue in try_activate'
         1: from C:/opscode/chefdk/embedded/lib/ruby/2.5.0/rubygems/specification.rb:1438:in `activate'
C:/opscode/chefdk/embedded/lib/ruby/2.5.0/rubygems/specification.rb:2325:in `raise_if_conflicts': Unable to activate knife-azure-1.8.0, because azure_mgmt_resources-0.17.1 conflicts with azure_mgmt_resources (= 0.9.0), azure_mgmt_network-0.17.3 conflicts with azure_mgmt_network (= 0.9.0) (Gem::ConflictError)

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@msystechnologies.com>

### Description
Upgraded azure-sdk libraries i.e.
azure_mgmt_resources
azure_mgmt_compute
azure_mgmt_storage
azure_mgmt_network

This will also resolve the conflict error mentioned in #490 

Tested manually with all knife azurerm server commands.
### Issues Resolved
#490

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/azure/resource_management/ARM_interface.rb [hunk 1]
```diff
@@ -29,17 +29,17 @@ class ARMInterface < AzureInterface
...
 
-      include Azure::ARM::Storage
-      include Azure::ARM::Storage::Models
+      include Azure::Storage::Mgmt::V2018_07_01
+      include Azure::Storage::Mgmt::V2018_07_01::Models
 
-      include Azure::ARM::Network
-      include Azure::ARM::Network::Models
+      include Azure::Network::Mgmt::V2018_08_01
+      include Azure::Network::Mgmt::V2018_08_01::Models
```
Comment (btm) [inline lib/azure/resource_management/ARM_interface.rb]: It looks like this wasn't added until https://github.com/Azure/azure-sdk-for-ruby/commit/8fb678ddaad195b1da2d31b073d82379b000cc66 which means your dependency on `azure_mgmt_network` should be `"~> 0.17.3"

https://rubygems.org/gems/azure_mgmt_network/versions/0.17.3

File: lib/azure/resource_management/ARM_interface.rb [hunk 2]
```diff
@@ -29,17 +29,17 @@ class ARMInterface < AzureInterface
...
 
-      include Azure::ARM::Compute
-      include Azure::ARM::Compute::Models
+      include Azure::Compute::Mgmt::V2018_06_01
+      include Azure::Compute::Mgmt::V2018_06_01::Models
 
-      include Azure::ARM::Storage
-      include Azure::ARM::Storage::Models
+      include Azure::Storage::Mgmt::V2018_07_01
+      include Azure::Storage::Mgmt::V2018_07_01::Models
```
Comment (btm) [inline lib/azure/resource_management/ARM_interface.rb]: https://github.com/Azure/azure-sdk-for-ruby/commit/5effef547df2362daf8a1a12021de37598d63e2a

https://rubygems.org/gems/azure_mgmt_storage/versions/0.17.2

File: lib/azure/resource_management/ARM_interface.rb [hunk 3]
```diff
@@ -29,17 +29,17 @@ class ARMInterface < AzureInterface
...
 
-      include Azure::ARM::Resources
-      include Azure::ARM::Resources::Models
+      include Azure::Resources::Mgmt::V2018_05_01
+      include Azure::Resources::Mgmt::V2018_05_01::Models
 
-      include Azure::ARM::Compute
-      include Azure::ARM::Compute::Models
+      include Azure::Compute::Mgmt::V2018_06_01
+      include Azure::Compute::Mgmt::V2018_06_01::Models
```
Comment (btm) [inline lib/azure/resource_management/ARM_interface.rb]: https://github.com/Azure/azure-sdk-for-ruby/commit/096d78b680d1d5ac1986947ce16aeef122273686

https://rubygems.org/gems/azure_mgmt_compute/versions/0.18.0

File: lib/azure/resource_management/ARM_interface.rb [hunk 4]
```diff
@@ -29,17 +29,17 @@ class ARMInterface < AzureInterface
       include Azure::ARM::ARMDeploymentTemplate
       include Azure::ARM::VnetConfig
 
-      include Azure::ARM::Resources
-      include Azure::ARM::Resources::Models
+      include Azure::Resources::Mgmt::V2018_05_01
+      include Azure::Resources::Mgmt::V2018_05_01::Models
```
Comment (btm) [inline lib/azure/resource_management/ARM_interface.rb]: https://github.com/Azure/azure-sdk-for-ruby/commit/096d78b680d1d5ac1986947ce16aeef122273686

https://rubygems.org/gems/azure_mgmt_resources/versions/0.17.0

**Comment by Vasu1105** [comment]
@btm I have pinned the versions could you please have a look at this ?

File: knife-azure.gemspec [hunk 5]
```diff
@@ -21,10 +21,10 @@ Gem::Specification.new do |s|
 
   s.add_dependency "nokogiri", ">= 1.5.5"
   s.add_dependency "knife-windows", "~> 1.0"
-  s.add_dependency "azure_mgmt_resources", ">= 0.9.0"
-  s.add_dependency "azure_mgmt_compute", ">= 0.9.0"
-  s.add_dependency "azure_mgmt_storage", ">= 0.9.0"
-  s.add_dependency "azure_mgmt_network", ">= 0.9.0"
+  s.add_dependency "azure_mgmt_resources", "~> 0.17.0"
```
Comment (tas50) [inline knife-azure.gemspec]: @Vasu1105 Can we make these all >= requirements so we don't introduce constraints with other things in our stack that use these same libraries

Comment (Vasu1105) [reply knife-azure.gemspec]: @tas50 did the changes.

**Comment by tas50** [comment]
Ping @stuartpreston

**Comment by stuartpreston** [comment]
https://github.com/Azure/azure-sdk-for-ruby/issues/1968 has caused us to [pin azure_mgmt_network to `= 0.18.0` in kitchen-azurerm](https://github.com/test-kitchen/kitchen-azurerm/commit/7c1b400d43424aa84529f247c0b9ebf0b48865b5), along with azure_mgmt_resources to `= 0.17.0`. If we can do the same here and create a chore to monitor that issue for resolution then we will ship fewer gems along with Chef/ChefDK/Workstation.

**Comment by Vasu1105** [comment]
@stuartpreston please have a look I have updated the gemspec

**Comment by stuartpreston** [comment]
This looks good but I'm just waiting for confirmation from the MSFT team as I think they are going to fix their release, which will allow us to unpin this.

**Comment by stiller-leser** [comment]
Just as a FYI: I opened https://social.msdn.microsoft.com/Forums/en-US/862b0824-b335-4559-9463-21c4abe5e7a9/azure-sdk-for-ruby-azuremgmtnetwork-0181-uses-old-versions-of-networkinterfaces-and?forum=AzureAPIApps in the Azure forums as well, since there doesn't seem to be any reaction at all in their GitHub repository.

**Comment by stuartpreston** [comment]
azure_mgmt_network is now 0.18.2, please can we make the dependencies take this as as a minimum version?  I'll do the same with kitchen-azurerm and then we can coordinate a release for the ChefDK/Workstation.

**Comment by stuartpreston** [comment]
@Vasu1105 Kitchen-AzureRM now has the following dependencies, please update knife-azure deps for compatibility: https://github.com/test-kitchen/kitchen-azurerm/blob/master/kitchen-azurerm.gemspec#L16-L17

**Comment by tas50** [comment]
We should get this released so we can ship it in the next DK

**Comment by kirantsp** [comment]
Hi all, please let me know when this would get released. I have a similar issue (#24474) which is referenced in #490 - Regards, Kiran.

**Pull request merged**